src-tool/Pact/Analyze/Translate.hs

```diff
@@ -1516,6 +1516,13 @@ translateNode astNode = withAstContext astNode $ case astNode of
 
   AST_NFun node "list" _ -> throwError' $ DeprecatedList node
 
+  -- rs (2023-05-10): Note, we handle the empty list properly (see #1182)
+  -- as the element type is otherwise inferred as `Any`.
+  AST_List node [] -> translateType node >>= \case
+    EType retTy -> case retTy of
```

Comment:
Why is it called `retTy` here? What is `node` at this point?

Reply:
`retTy` is the return type of the translation step, i.e. `SList SStr`.  I will change it to `listTy`.

The `node` is of type `Node` from the Typecheck module, see [here](https://github.com/kadena-io/pact/blob/master/src/Pact/Types/Typecheck.hs#L290). For the particular example above, its `TyList {_tyListType = TyPrim TyString}`.